wrappers/rust/icicle-curves/icicle-curve-template/src/poseidon/mod.rs

```diff
@@ -0,0 +1,22 @@
+use crate::curve::{ScalarCfg, ScalarField};
+
+use icicle_core::error::IcicleResult;
+use icicle_core::impl_poseidon;
+use icicle_core::poseidon::{Poseidon, PoseidonConfig, PoseidonConstants};
+use icicle_core::traits::IcicleResultWrap;
+use icicle_cuda_runtime::device_context::DeviceContext;
+use icicle_cuda_runtime::error::CudaError;
+use icicle_cuda_runtime::memory::HostOrDeviceSlice;
+
+use core::mem::MaybeUninit;
+
+impl_poseidon!("<CURVE>", <CURVE>, ScalarField, ScalarCfg);
+
+#[cfg(test)]
+pub(crate) mod tests {
+    use crate::curve::ScalarField;
+    use icicle_core::impl_poseidon_tests;
+    use icicle_core::poseidon::tests::*;
+
+    impl_poseidon_tests!(ScalarField);
```

Comment:
What about the introduction of the new `impl_poseidon_custom_config_test` macro?

Reply:
it's purposely skipped